### PR TITLE
build: include esm and cjs builds for better compatibility

### PR DESCRIPTION
**Pull request by napei**
I recently started using this library in Angular, which like many other build optimisers prefers ESModules instead of CommonJS to be able to tree-shake the source code, and be overall more efficient. This change essentially specifies a CommonJS build and an ESM build separately to be used by whichever.

I have not tested this change rigorously as I don't work with npm packages too much.

Perhaps this is something that can be investigated?

### CODE REVIEW

**Comment by pcafstockf** [comment]
Anything I can do to help the PR along?

**Comment by beaugunderson** [comment]
i missed this one entirely--will take a peek; ping me in a day if i don't report back :)

**Comment by napei** [comment]
> i missed this one entirely--will take a peek; ping me in a day if i don't report back :)

Awesome! I should have made an issue to go along with it. I'll rebase the branch and get it fixed again.

**Comment by napei** [comment]
> ping me in a day if i don't report back :)

@beaugunderson just checking in 👋

File: package.json [hunk 1]
```diff
@@ -43,8 +44,7 @@
     "node": ">= 10"
   },
   "files": [
-    "dist/**/*.js",
-    "dist/**/*.ts"
+    "dist/"
```
Comment (beaugunderson) [inline package.json:47]: what files weren't match by the two includes?

File: package.json [hunk 2]
```diff
@@ -10,15 +10,16 @@
...
   "scripts": {
     "docs": "documentation build --github --output docs --format html ./ip-address.js",
-    "prepublishOnly": "tsc",
+    "build": "tsc -p tsconfig.json && tsc -p tsconfig-cjs.json",
+    "prepublishOnly": "npm run build",
     "release": "release-it",
-    "test-ci": "nyc mocha",
-    "test": "mocha",
-    "watch": "mocha --watch"
+    "test-ci": "nyc jest --config jest.config.js --ci",
```
Comment (beaugunderson) [inline package.json:20]: would prefer to continue using `mocha`; what issues did you have with it? i'm happy to fix that aspect

File: package.json [hunk 3]
```diff
@@ -10,15 +10,16 @@
   "version": "7.1.0",
   "author": "Beau Gunderson <beau@beaugunderson.com> (https://beaugunderson.com/)",
   "license": "MIT",
-  "main": "dist/ip-address.js",
-  "types": "dist/ip-address.d.ts",
+  "main": "dist/cjs/ip-address.js",
+  "module": "dist/esm/ip-address.js",
```
Comment (beaugunderson) [inline package.json:14]: looks like we lost the `types` declaration here?

File: package.json (hunk 2 above)
Comment (napei) [reply package.json:20]: Ran into mocha not liking the module config in package.json and tried for about 20 minutes then gave up. Happy for it to be fixed however and really I shouldn't have put it into the PR.

File: package.json (hunk 1 above)
Comment (napei) [reply package.json:47]: Felt it was more semantic to say that the `dist` folder is the package output in its entirety. Also `*.js.map` was missing so just using `dist/` covered everything.

File: package.json (hunk 3 above)
Comment (napei) [reply package.json:14]: Fixed with de9c6c2

File: package.json (hunk 2 above)
Comment (napei) [reply package.json]: I've reverted that commit with 742410e so you can see how it was working beforehand.

**Comment by napei** [comment]
@beaugunderson ping 🔔

Just checking in 😄

File: package.json (hunk 1 above)
Comment (beaugunderson) [reply package.json:48]: gotcha, the map files are what I was looking for--I think ok to use `dist/` but have been burnt in the past by adding large files by accident :)

**Comment by beaugunderson** [comment]
@napei thank you! appreciate the ping 🙏 gonna look at the test failures now and will get this merged

**Comment by beaugunderson** [comment]
thanks @napei, will release this as 8.0.0 shortly and include some dependency updates as well

**Pull request closed without merging**